Document routes API (#6604)
plone/volto: packages/registry/docs/how-to-guides/register-routes.md

# Register new routes

The `config.registerRoute` method adds a route to the configuration registry.
It saves the routes in the `config.routes` key in the configuration object.

> You should provide one route at a time and must have the following shape:

Reviewer: ```suggestion
You should provide one route at a time.
Each route must have the following data structure.
```

The `type`, `path`, and `file` are mandatory.
The `type` key specifies the route type to create, specifically one of `route`, `index`, `layout`, or `prefix`.
The type `route` can contain nested routes.

```{info}

> The routes registered with this method are intended to be React Router 7 compliant routes.

Reviewer: ```suggestion
The routes registered with this method must be compliant with React Router 7 routes.
```

> They are meant to be loaded via a helper provided by `@plone/react-router` in an existing React Router 7 app.

Reviewer: ```suggestion
They are loaded by a helper provided by `@plone/react-router` in an existing React Router 7 app.
```
